Refactor deploy documentation (#2502)
astronomer/docs: astro/deploy-project-image.md

When the deploy completes, the **Docker Image** field in the Cloud UI are updated depending on the type of deploy you completed. 

- The **Docker Image** field displays a unique identifier generated by a Continuous Integration (CI) tool or a timestamp generated by the Astro CLI after you complete an image deploy. 

To confirm a deploy was successful, verify that the running versions of your Docker image and DAG bundle have been updated.

1. In the Cloud UI, select a Workspace and then select a Deployment.
2. Review the information in the **Docker Image** field to determine the Deployment code version.

:::info

If your Deployment has [DAG-only deploys](deploy-dags.md) enabled, the **DAG bundle version** field will also update with a timestamp. For Deployments where this feature is enabled, the Astro CLI deploys DAGs separately from the project image.

:::

## What happens during a project deploy

> When you deploy a project to Astro, your Astro project is built into a Docker image. This includes system-level dependencies, Python-level dependencies, and your `Dockerfile`. It does not include any of the metadata associated with your local Airflow environment, including task history and Airflow connections or variables that were set locally. This Docker image is then pushed to all containers running Apache Airflow on Astro. If the DAG-only deploy feature is enabled for your Deployment, the `/dags` directory is excluded from this Docker image and is pushed separately.

Reviewer: ```suggestion
When you deploy a project to Astro, your Astro project is built into a Docker image. This includes system-level dependencies, Python-level dependencies, and your `Dockerfile`. It does not include any of the metadata associated with your local Airflow environment, including task history and Airflow connections or variables that were set locally. This Docker image is then pushed to all containers running Apache Airflow on Astro. If the DAG-only deploy feature is enabled for your Deployment, the `dags` directory is excluded from this Docker image and is pushed separately.
```

![Deploy code](/img/docs/deploy-architecture.png)

> With the exception of the Airflow webserver and some Celery workers, Kubernetes gracefully terminates all containers during this process. This forces them to restart and begin running your latest code.

Reviewer: ```suggestion
Astro gracefully terminates all containers during this process, except Airflow webserver and some Celery workers, if they are executing tasks. The new containers will come up your latest code and existing Celery workers will finish the tasks before being terminated. See the example scenario below to understand how Astro handles this.
```

:::tip

If you want to force long-running tasks to terminate sooner than 24 hours, specify an [`execution_timeout`](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/tasks.html#timeouts) in your DAG's task definition.

:::

## Deploy a prebuilt Docker image

By default, running `astro deploy` with the Astro CLI builds your Astro project into a Docker image and deploys it to Astro. In some cases, you might want to skip the build step and deploy a prebuilt Docker image instead.

Deploying a prebuilt Docker image allows you to:

- Test a single Docker image across Deployments instead of rebuilding it each time.
- Reduce the time it takes to deploy. If your Astro project has a number of packages that take a long time to install, it can be more efficient to build it separately.
- Specify additional mounts and arguments in your project, which is required for setups such as [installing Python packages from private sources](develop-project.md#install-python-packages-from-private-sources).

To deploy your Astro project as a prebuilt Docker image:

> 1. Run `docker build` from an Astro project directory or specify the command in a CI/CD pipeline. This Docker image must be based on Astro Runtime and be available in a local Docker registry. If you run this command on an Apple M1 computer or on a computer with an ARM64 processor, you must specify `--platform=linux/amd64` or else the deploy will fail. Astro Deployments require an AMD64-based image and do not support ARM64 architecture.

Reviewer: ```suggestion
1. Run `docker build` from an Astro project directory or specify the command in a CI/CD pipeline. This Docker image must be based on Astro Runtime and be available in a local Docker registry. If you run this command on an Apple M1 computer or on a computer with an ARM64 processor, you must specify `--platform=linux/amd64` in the `FROM` statement of Dockerfile or else the deploy will fail. Astro Deployments require an AMD64-based image and do not support ARM64 architecture.
```

Author: `--platform` is a command-level flag, how would you specify this in a Dockerfile? Shouldn't this be "in the command"? https://docs.docker.com/engine/reference/commandline/build/#options
